test/unit/proxy_api/proxy_exception_test.rb

```diff
@@ -0,0 +1,26 @@
+require 'test_helper'
+
+class ProxyApiExceptionTest < ActiveSupport::TestCase
+  test 'exception response body is added if available' do
+    fake_exception = OpenStruct.new(:response => 'useful message from proxy',
```

Comment:
If you change this message here and in the expectation, e.g. s/message/response/ then the test will fail:

    Expected /ERF.*ProxyException.*unable to do something.*useful response from proxy.*/ to match "ERF12-2902 [ProxyAPI::ProxyException]: unable to do something ([OpenStruct]: useful message from proxy) for proxy fakeurl".

The response isn't in the message.